lib/ecto/query/builder/select.ex

```diff
@@ -24,6 +24,12 @@ defmodule Ecto.Query.Builder.Select do
 
   """
   @spec escape(Macro.t, Keyword.t, Macro.Env.t) :: {Macro.t, {list, %{}}}
+  def escape(atom, _vars, _env) when is_atom(atom) and not is_boolean(atom) do
```

Comment:
Unfortunately I could not find any better place than putting the check here. Would we prefer to extracting this into a helper function instead?

Reply:
Looks great but we also need to check it is not nil.

Reply:
Oops right!